mypy/meet.py

```diff
@@ -645,7 +646,18 @@ def are_tuples_overlapping(
 
     if len(left.items) != len(right.items):
         return False
-    return all(is_overlapping(l, r) for l, r in zip(left.items, right.items))
+    if not all(is_overlapping(l, r) for l, r in zip(left.items, right.items)):
+        return False
+
+    # Check that the tuples aren't from e.g. different NamedTuples.
+    if is_named_instance(right.partial_fallback, "builtins.tuple") or is_named_instance(
+        left.partial_fallback, "builtins.tuple"
+    ):
+        return True
+    else:
+        return is_subtype(left.partial_fallback, right.partial_fallback) or is_subtype(
+            right.partial_fallback, left.partial_fallback
+        )
```

Comment:
I don't really get the difference between `is_overlapping` and two `is_subtype` calls but I suppose I should use `is_overlapping` here instead?